src/main/java/com/minecolonies/rankup/util/AccountingUtils.java

```diff
@@ -0,0 +1,524 @@
+                accConfig.playerData.get(uuid).playerName = name;
+                accConfig.save();
+            }
+        }
+        catch (SQLException | NoModuleException e)
+        {
+            e.printStackTrace();
+        }
+    }
+
+    public void updatePlayerJoinDate(final UUID uuid, final String date)
+    {
+        try
+        {
+            if (plugin.getModuleContainer().isModuleLoaded(DatabaseModule.ID))
+            {
+                updatePlayerAttribute(uuid, JOIN_DATE_COLUMN, date, true);
+            }
+            else
+            {
+
+                final AccountConfigData accConfig = getAccountConfig();
+                accConfig.playerData.get(uuid).joinDate = CommonUtils.dateFormat(date);
+                accConfig.save();
+            }
+        }
+        catch (SQLException | NoModuleException e)
+        {
+            e.printStackTrace();
+        }
+    }
+
+    public void updatePlayerLastDate(final UUID uuid, final String date)
+    {
+        try
+        {
+            if (plugin.getModuleContainer().isModuleLoaded(DatabaseModule.ID))
+            {
+                updatePlayerAttribute(uuid, LAST_JOIN_COLUMN, date, true);
+            }
+            else
+            {
+                final AccountConfigData accConfig = getAccountConfig();
+                accConfig.playerData.get(uuid).lastVisit = CommonUtils.dateFormat(date);
+                accConfig.save();
+            }
+        }
+        catch (SQLException | NoModuleException e)
+        {
+            e.printStackTrace();
+        }
+    }
+
+    public void updatePlayerTime(final UUID uuid, final int time)
+    {
+        try
+        {
+            if (plugin.getModuleContainer().isModuleLoaded(DatabaseModule.ID))
+            {
+                updatePlayerAttribute(uuid, TIME_PLAYED_COLUMN, time);
+            }
+            else
+            {
+                final AccountConfigData accConfig = getAccountConfig();
+                accConfig.playerData.get(uuid).timePlayed = time;
+                accConfig.save();
+            }
+        }
+        catch (SQLException | NoModuleException e)
+        {
+            e.printStackTrace();
+        }
+    }
+
+    public int addPlayerTime(final UUID uuid, final int time)
+    {
+        try
+        {
+            if (plugin.getModuleContainer().isModuleLoaded(DatabaseModule.ID))
+            {
+                final int timeToAdd = getPlayerTime(uuid) + time;
+                updatePlayerTime(uuid, timeToAdd);
+                return timeToAdd;
+            }
+            else
+            {
+                final AccountConfigData accConfig = getAccountConfig();
+                final int timeToAdd = accConfig.playerData.get(uuid).timePlayed + time;
+                accConfig.playerData.get(uuid).timePlayed = timeToAdd;
+                accConfig.save();
+                return timeToAdd;
+            }
+        }
+        catch (NoModuleException e)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1166)